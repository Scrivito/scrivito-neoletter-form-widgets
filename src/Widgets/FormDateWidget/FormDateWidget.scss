.scrivito-neoletter-form-widgets .form-date {
  .date-title {
    p {
      display: inline;
    }
  }
}
