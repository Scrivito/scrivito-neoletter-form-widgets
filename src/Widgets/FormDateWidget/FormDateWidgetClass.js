import * as Scrivito from "scrivito";

export const FormDateWidget = Scrivito.provideWidgetClass("FormDateWidget", {
  attributes: {
    title: "string",
    helpText: "html",
    required: "boolean",
    customFieldName: "string",
    dateType: [
      "enum",
      {
        values: ["date", "datetime-local"],
      },
    ],
  },
});
