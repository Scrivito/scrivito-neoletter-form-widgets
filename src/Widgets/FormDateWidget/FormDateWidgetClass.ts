import * as Scrivito from "scrivito";

export const FormDateWidget = Scrivito.provideWidgetClass("FormDateWidget", {
  attributes: {
    title: "html",
    helpText: "html",
    required: "boolean",
    customFieldName: "string",
    dateType: [
      "enum",
      {
        values: ["date", "datetime-local"]
      }
    ]
  }
});
