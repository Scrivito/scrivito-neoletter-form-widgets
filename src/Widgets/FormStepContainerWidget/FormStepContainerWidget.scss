// Styles for FormContainerWidget
.scrivito-neoletter-form-widgets.form-container-widget {

  .dropdown-container,
  .select-container {
    .select-title {
      display: flex;
      align-items: baseline;
    }
  }

  .text-super {
    vertical-align: super;
  }

  .text-mandatory {
    font-weight: 700;
    color: var(--bs-danger, #f03a47); //$secondary;
    cursor: default;
  }

  .rating-icon.bi {
    color: grey;
  }

  .winnie-the-pooh {
    opacity: 0;
    position: absolute;
    top: 0;
    left: 0;
    height: 0;
    width: 0;
    z-index: -1;
  }

  .form-buttons {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;

    .step-counter {
      width: 100%;
      display: flex;
      flex-direction: row;
      justify-content: center;
      align-items: center;
    }

    .review-button {
      margin-right: 1em;
    }
  }

  .hide {
    display: none;
  }

  &.form-border {
    border: black 1px dotted;
    padding: 0.5em;
  }

  .captcha-container {
    display: flex;

    &.left {
      justify-content: left;
    }

    &.center {
      justify-content: center;
    }

    &.right {
      justify-content: right;
    }
  }

  .missing-site-key {
    border: #f03a47 1px solid;
    padding: 1em 0;
  }

  // popover
  .mandatory-container,
  .helptext-container {
    position: relative;
    display: inline-block;

    .form-popover-container {
      position: absolute;
      inset: auto auto 0px 0px;
      transform: translate(-50px, -30px);
      background-color: #fff;
      border: 1px solid #ccc;
      border-radius: 0.5rem;
      box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
      z-index: 1000;
      max-width: 276px;
      width: max-content;

      .form-popover-body {
        font-size: 14px;
        color: #677788;
        padding: 1rem;
        font-style: normal;
        font-weight: 400;
        line-height: 1.5;
        word-wrap: break-word;

        p {
          margin: 0;
        }
      }
    }
  }

  // review modal
  .form-modal {
    position: fixed;
    top: 0;
    left: 0;
    z-index: 1060;
    width: 100%;
    height: 100%;
    overflow-x: hidden;
    overflow-y: auto;
    outline: 0;
    background-color: #212529;

    .modal-dialog {
      width: 80%;
      max-width: 800px;
      margin: 1.75rem auto;

      @keyframes animatetop {
        from {
          top: -300px;
          opacity: 0
        }

        to {
          top: 0;
          opacity: 1
        }
      }

      .modal-content {
        animation: animatetop 0.4s;
        position: relative;
        display: flex;
        flex-direction: column;
        width: 100%;
        pointer-events: auto;
        background-color: #fff;
        background-clip: padding-box;
        border: 1px solid rgba(0, 0, 0, .2);
        border-radius: .3rem;
        outline: 0;

        .modal-header {
          display: flex;
          flex-shrink: 0;
          align-items: center;
          justify-content: center;
          padding: 1rem 1rem;
          border-bottom: 1px solid #dee2e6;
          border-top-left-radius: calc(.3rem - 1px);
          border-top-right-radius: calc(.3rem - 1px);

          h4 {
            margin-bottom: 0;
            line-height: 1.5;
            font-family: "Segoe UI", Arial, sans-serif;
            font-weight: 500;
            margin: 10px 0;
          }
        }

        .modal-body.form-review {
          height: 60vh;
          overflow: auto;
          position: relative;
          flex: 1 1 auto;
          padding: 1rem;
          pointer-events: auto;
          word-break: break-word;

          div.step-review-container {
            &:not(:first-child) {
              .step-review {
                margin-top: 1em !important;
              }
            }
          }

          .review-item-title {
            color: #9e9e9e;
            margin-bottom: 0;
            margin-top: 1em;
            font-size: 1em;
            font-weight: normal;
          }

          .review-item-value {
            //  margin-top: 0.3em;
            font-weight: normal;
            white-space: pre-wrap;
            max-width: 100vw;
            overflow-x: auto;
          }
        }

        .modal-footer {
          display: flex;
          flex-wrap: wrap;
          flex-shrink: 0;
          align-items: center;
          justify-content: flex-end;
          padding: .75rem;
          border-top: 1px solid #dee2e6;
          border-bottom-right-radius: calc(.3rem - 1px);
          border-bottom-left-radius: calc(.3rem - 1px);

          button {
            color: #fff;
            background-color: var(--bs-primary, #5c9dcd);
            border-color: var(--bs-primary, #5c9dcd);
            margin: .25rem;
            padding: .375rem .75rem;
            font-size: 1rem;
            border-radius: .25rem;

            &:hover {
              background-color: var(--bs-btn-hover-bg, #395aa5);
            }
          }
        }
      }
    }
  }

  // Bootstrap Icons - Sizes Add-on
  .bi {
    &.bi-1x {
      font-size: 1em;
      line-height: 1;
    }

    &.bi-default {
      font-size: 1.3em;
      line-height: 1;
    }

    &.bi-2x {
      font-size: 2em;
      line-height: 1;
    }

    &.bi-3x {
      font-size: 3em;
      line-height: 1;
    }

    &.bi-4x {
      font-size: 4em;
      line-height: 1;
    }

    &.bi-5x {
      font-size: 5em;
      line-height: 1;
    }
  }

  @keyframes rotate-icon {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }

  .rotate-icon {
    display: inline-block;
    animation: rotate-icon 2s linear 0s infinite;
  }
}

.bg-dark-image,
.bg-brand-primary,
.bg-brand-secondary,
.bg-grey,
.bg-greydark,
.bg-greymiddle {

  .missing-id,
  .missing-site-key,
  .scrivito-neoletter-form-widgets.form-container-widget .linear-scale-container,
  .scrivito-neoletter-form-widgets.form-container-widget .reset-label span,
  .scrivito-neoletter-form-widgets.form-container-widget label,
  .scrivito-neoletter-form-widgets.form-container-widget .text-mandatory,
  .scrivito-neoletter-form-widgets.form-container-widget .bi,
  .scrivito-neoletter-form-widgets.form-container-widget .text-super,
  .card.card-theme .scrivito-neoletter-form-widgets.form-container-widget label,
  .card.card-theme .scrivito-neoletter-form-widgets.form-container-widget .text-mandatory,
  .card.card-theme .scrivito-neoletter-form-widgets.form-container-widget .bi {
    color: #fff;
  }

  // hardcoded for now
  .card .scrivito-neoletter-form-widgets.form-container-widget label {
    color: #495054
      /*$theme-grey */
      !important;
  }

  .card .scrivito-neoletter-form-widgets.form-container-widget .text-mandatory {
    color: red !important;
  }

  .card .scrivito-neoletter-form-widgets.form-container-widget .bi {
    color: #ccc !important;
  }
}

.bg-dark-image,
.bg-brand-primary,
.bg-brand-secondary,
.bg-grey,
.bg-greydark,
.bg-greymiddle {
  .scrivito-neoletter-form-widgets.form-container-widget .reset-label:hover {
    background-color: unset;
  }
}

.scrivito-neoletter-form-widgets.missing-id {
  display: flex;
  justify-content: center;
}

// FormIdComponent tab 
.neoletter-form-id-tab-container {
  .scrivito_detail_content {
    border-top: 2px solid rgba(0, 0, 0, 0.1);
    margin: 0;
    padding: 15px;
    font-family: Helvetica Neue, Helvetica, Arial, sans-serif;

    .scrivito_detail_label {
      display: block;
      padding: 0;
      padding-bottom: 4px;
      width: 100%;

      span {
        padding: 7px 3px 6px 3px;
        color: #444;
        display: block;
        font-size: 13px;
        font-weight: bold;
        font-family: "Manrope", sans-serif;
      }

      &.scrivito_error {

        i,
        span {
          color: #c04e4e !important;
        }

        .scrivito_icon_error {
          float: left;
          padding: 9px 3px;
          font-size: 11px;
          font-style: normal;
          font-weight: normal;
          line-height: 13px !important;
          vertical-align: middle;
          text-decoration: none;
          position: relative;
          -webkit-font-smoothing: antialiased;
          -moz-osx-font-smoothing: grayscale;
          -webkit-user-select: none;
          -moz-user-select: none;
          -ms-user-select: none;
          -o-user-select: none;
          user-select: none;
        }
      }
    }

    .content_property_input {
      margin-bottom: 5px;
      box-shadow: 0 0 3px rgba(0, 0, 0, .2);
      border-radius: 3px;
      background: white;
      color: #555;
      display: block;
      font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
      font-size: 14px;
      font-style: normal;
      margin-bottom: 5px;
      padding: 0 8px;
      min-height: 30px;
      line-height: 30px;
      word-break: break-all;

      &.scrivito_error {
        box-shadow: 0 0 0 2px #c04e4e inset;
      }
    }

    .scrivito_validation_notice {
      border-left-width: 7px !important;
      padding: 5px 10px;
      font-size: 11px;
      min-height: 30px;
      line-height: 19px;
      font-weight: normal;
      margin-bottom: 5px;
      box-shadow: 0 0 3px rgba(0, 0, 0, .2);
      border-radius: 3px;
      background: rgba(255, 255, 255, .6);

      &.scrivito_error {
        border-left: 2px solid #c04e4e;
        color: #666;
        box-shadow: 0 0 0 2px #c04e4e inset;
      }
    }

    .scrivito_notice_body {
      color: #666;
      font-weight: bold;
      font-size: 12px;
      padding: 0 0 10px;
    }

    .scrivito_button {
      display: inline-block;
      color: #555;
      font-size: 14px;
      font-weight: normal;
      line-height: 14px;
      text-align: center;
      padding: 8px 10px;
      margin: 0 10px 10px 0;
      position: relative;
      z-index: 1;
      vertical-align: middle;
      min-width: 90px;
      min-height: 14px;
      font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
      background: #fcfcfc;
      border-radius: 3px;
      -webkit-user-select: none;
      -moz-user-select: none;
      -ms-user-select: none;
      -o-user-select: none;
      user-select: none;
      text-decoration: none;

      &.scrivito_disabled {
        color: #aaa !important;
        cursor: not-allowed;
        background: #eee;
      }

      &.scrivito_blue {
        background: #426698 !important;
        color: #fff !important;
      }
    }
  }

  &.scrivito_dark {
    .scrivito_detail_label span {
      color: #eee;
      text-shadow: 0 1px 1px rgba(0, 0, 0, .3), 0 0 4px rgba(0, 0, 0, .2);
    }

    .content_property_input {
      background: rgba(222, 222, 222, .2);
      color: #eee;
    }

    .scrivito_validation_notice {
      color: #eee !important;
      background: rgba(222, 222, 222, 0.2);
    }
  }
}