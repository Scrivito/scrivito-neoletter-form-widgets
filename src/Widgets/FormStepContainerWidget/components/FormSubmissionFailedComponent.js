import * as React from "react";

export const FormSubmissionFailed = ({ submissionFailureText }) => {
  return (
    <div className="scrivito-form-widgets form-container-widget text-center">
      <i className="fa fa-exclamation-triangle fa-2x" aria-hidden="true"></i>{" "}
      <span className="text-super">{submissionFailureText}</span>
    </div>
  );
};
