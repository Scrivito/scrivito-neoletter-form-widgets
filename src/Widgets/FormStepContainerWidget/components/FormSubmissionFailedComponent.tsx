import * as React from "react";

interface FormSubmissionFailedProps {
  submissionFailureText: string;
}
export const FormSubmissionFailed: React.FC<FormSubmissionFailedProps> = ({
  submissionFailureText
}) => {
  return (
    <div className="scrivito-neoletter-form-widgets form-container-widget text-center">
      <i
        className="bi bi-exclamation-triangle-fill bi-2x"
        aria-hidden="true"></i>{" "}
      <span className="text-super">{submissionFailureText}</span>
    </div>
  );
};
