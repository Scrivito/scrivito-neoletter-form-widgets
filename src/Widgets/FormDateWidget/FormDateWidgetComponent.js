import * as React from "react";
import * as Scrivito from "scrivito";
import { getFieldName } from "../FormContainerWidget/utils/getFieldName";
import { HelpText } from "../FormContainerWidget/components/HelpTextComponent";
import { Mandatory } from "../FormContainerWidget/components/MandatoryComponent";
import "./FormDateWidget.scss";

Scrivito.provideComponent("FormDateWidget", ({ widget }) => {
  const [value, setValue] = React.useState("");
  const onChangeValue = (e) => {
    const d = new Date(e.target.value);
    setValue(d.toISOString());
  };
  return (
    <div className="form-date mb-3">
      <div className="date-title">
        <span className="text-super">{widget.get("title")}</span>
        {widget.get("required") && <Mandatory />}
        {widget.get("helpText") && <HelpText widget={widget} />}
      </div>

      <input
        onChange={onChangeValue}
        className="datepicker"
        type={widget.get("dateType")}
        required={widget.get("required")}
      ></input>
      <input type="hidden" name={getFieldName(widget)} value={value}></input>
    </div>
  );
});
