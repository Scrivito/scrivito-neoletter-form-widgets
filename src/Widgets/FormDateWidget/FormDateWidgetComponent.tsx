import * as React from "react";
import * as Scrivito from "scrivito";
import { getFieldName } from "../FormStepContainerWidget/utils/getFieldName";
import { HelpText } from "../FormStepContainerWidget/components/HelpTextComponent";
import { Mandatory } from "../FormStepContainerWidget/components/MandatoryComponent";
import { FormDateWidget } from "./FormDateWidgetClass";
import "./FormDateWidget.scss";

Scrivito.provideComponent(FormDateWidget, ({ widget }) => {
  const [value, setValue] = React.useState("");
  const onChangeValue = (e: React.BaseSyntheticEvent) => {
    const d = new Date(e.target.value);
    setValue(d.toISOString());
  };
  return (
    <div className="form-date mb-3">
      <div className="date-title">
        <span className="text-super">{widget.get("title")}</span>
        {widget.get("required") && <Mandatory />}
        {widget.get("helpText") && <HelpText widget={widget} />}
      </div>

      <input
        onChange={onChangeValue}
        className="datepicker"
        type={widget.get("dateType")!}
        required={widget.get("required")}></input>
      <input
        type="hidden"
        className="show-in-review"
        name={getFieldName(widget)}
        value={value}></input>
    </div>
  );
});
