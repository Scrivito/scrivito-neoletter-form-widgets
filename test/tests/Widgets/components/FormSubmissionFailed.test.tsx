import * as React from "react";
import * as Scrivito from "scrivito";
import { screen, render } from "@testing-library/react";
import { FormSubmissionFailed } from "../../../../src/Widgets/FormStepContainerWidget/components/FormSubmissionFailedComponent";
import renderer from "react-test-renderer";
import { DummyWidget } from "../../../helpers/dummyWidget";
import { Widget } from "scrivito";
import { FormDateWidget } from "../../../../src/Widgets/FormDateWidget/FormDateWidgetClass";
import { FormStepContainerWidget } from "../../../../src/Widgets/FormStepContainerWidget/FormStepContainerWidgetClass";
import "../../../../src/Widgets/FormStepContainerWidget/FormStepContainerWidgetComponent";
import "../../../../src/Widgets/FormDateWidget/FormDateWidgetComponent";
import PageRenderer from "../../../helpers/pageRenderer";

Scrivito.configure({ tenant: "inMemory" });
const submissionFailureText = "Submission failed!";

const formProps = {
  formId: "test-id",
  failedMessage: "failed",
  submittedMessage: "submitted",
  submittingMessage: "submitting",
  hiddenFields: [],
  formType: "single-step",
  steps: [],
  forwardButtonText: "forward",
  backwardButtonText: "back",
  submitButtonText: "submit",
  showBorder: false,
  showCaptcha: false,
  captchaAlignment: "center",
  showReview: true,
  includeEmptyAnswers: false,
  showStepsInReview: false,
  showReviewHeader: false,
  showReviewFooter: false,
  reviewButtonText: "Review",
  reviewHeaderTitle: "Review header",
  reviewCloseButtonText: "Close",
  singleSubmitButtonAlignment: "left",
  previewSubmittingMessage: false,
  previewFailedMessage: false,
  previewSubmittedMessage: false,
  submittingMessageType: "default",
  submittedMessageType: "default",
  failedMessageType: "default",
  submittedMessageWidgets: [],
  submittingMessageWidgets: [],
  failedMessageWidgets: [],
  showRetryButton: false,
  retryButtonText: "retry"
};

const pageRenderer = new PageRenderer();

describe("FormSubmissionFailed", () => {
  const widgetProps = {
    submissionFailureText: submissionFailureText,
    type: "default",

    showRetryButton: false,
    retryButtonText: "Retry",
    onReSubmit: () => { },
    buttonAlignment: "text-center"
  }
  const widget = new DummyWidget(widgetProps) as unknown as Widget;

  it("renders the exclamation icon and outputs the text", () => {
    const { container, getByText } = render(
      <FormSubmissionFailed
        {...widgetProps}
        widget={widget}
      />
    );

    const icon = container.querySelector(".bi-exclamation-triangle-fill");
    const textElement = getByText(submissionFailureText);
    expect(icon).toBeInTheDocument();
    expect(icon?.tagName.toLowerCase()).toBe("i");

    expect(textElement).toBeInTheDocument();
    expect(textElement.tagName.toLowerCase()).toBe("span");
  });

  it("renders the widget list without a retry button if edit mode is true", () => {
    jest.spyOn(Scrivito, "isInPlaceEditingActive").mockReturnValue(true);
    const submissionWidgets = [
      new FormDateWidget({
        customFieldName: "custom_date",
        title: "date",
        dateType: "date"
      })
    ];

    pageRenderer.render({
      body: [new FormStepContainerWidget({ ...formProps, failedMessageWidgets: submissionWidgets, failedMessageType: "widget-list", previewFailedMessage: true })]
    });

    const submissionContainer = document.querySelector(".form-submission-failed");
    const textContainer = document.querySelector(".text-center");
    const dateWidget = document.querySelector('input[type="date"]');
    const retryButton = document.querySelector("button");
    expect(textContainer).toBeNull();
    expect(dateWidget).toBeInTheDocument();
    expect(submissionContainer).toBeInTheDocument();
    expect(retryButton).not.toBeInTheDocument();
  });

  it("does not render if edit mode is false", () => {
    jest.spyOn(Scrivito, "isInPlaceEditingActive").mockReturnValue(false);

    const submissionWidgets = [
      new FormDateWidget({
        customFieldName: "custom_date",
        title: "date",
        dateType: "date"
      })
    ];

    pageRenderer.render({
      body: [new FormStepContainerWidget({ ...formProps, failedMessageWidgets: submissionWidgets, failedMessageType: "widget-list", previewFailedMessage: true })]
    });

    const textContainer = document.querySelector(".text-center");
    const dateWidget = document.querySelector('input[type="date"]');
    const form = document.getElementById("test-id");

    expect(textContainer).toBeNull();
    expect(dateWidget).not.toBeInTheDocument();
    expect(form).toBeInTheDocument();
  });

  it("does not render if preview is false and edit mode is true", () => {
    jest.spyOn(Scrivito, "isInPlaceEditingActive").mockReturnValue(false);

    const submissionWidgets = [
      new FormDateWidget({
        customFieldName: "custom_date",
        title: "date",
        dateType: "date"
      })
    ];

    pageRenderer.render({
      body: [new FormStepContainerWidget({ ...formProps, failedMessageWidgets: submissionWidgets, failedMessageType: "widget-list", previewFailedMessage: false })]
    });

    const textContainer = document.querySelector(".text-center");
    const spanElement = screen.queryByText(submissionFailureText);
    const dateWidget = document.querySelector('input[type="date"]');
    const form = document.getElementById("test-id");

    expect(textContainer).toBeNull();
    expect(spanElement).not.toBeInTheDocument();
    expect(dateWidget).not.toBeInTheDocument();
    expect(form).toBeInTheDocument();
  });

  it("shows the retry button ", () => {
    jest.spyOn(Scrivito, "isInPlaceEditingActive").mockReturnValue(true);

    const submissionWidgets = [
      new FormDateWidget({
        customFieldName: "custom_date",
        title: "date",
        dateType: "date"
      })
    ];

    pageRenderer.render({
      body: [new FormStepContainerWidget({ ...formProps, failedMessageWidgets: submissionWidgets, failedMessageType: "widget-list", previewFailedMessage: true, showRetryButton: true })]
    });

    const textContainer = document.querySelector(".text-center");
    const spanElement = screen.queryByText(submissionFailureText);
    const dateWidget = document.querySelector('input[type="date"]');
    const form = document.getElementById("test-id");
    const retryButton = screen.getByRole("button");
    expect(textContainer).not.toBeInTheDocument();
    expect(spanElement).not.toBeInTheDocument();
    expect(dateWidget).toBeInTheDocument();
    expect(form).not.toBeInTheDocument();
    expect(retryButton).toBeInTheDocument();
    expect(retryButton).toHaveTextContent("retry");
  });

  it("renders correctly", () => {
    const tree = renderer
      .create(
        <FormSubmissionFailed
          {...widgetProps}
          widget={widget}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
