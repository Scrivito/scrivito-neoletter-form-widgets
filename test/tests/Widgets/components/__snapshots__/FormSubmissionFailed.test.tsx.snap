// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSubmissionFailed renders correctly 1`] = `
<div
  className="scrivito-neoletter-form-widgets form-container-widget text-center"
>
  <i
    aria-hidden="true"
    className="bi bi-exclamation-triangle-fill bi-2x"
  />
   
  <span
    className="text-super"
  >
    Submission failed!
  </span>
</div>
`;
